### ISSUE TITLE
Sugestion #2 

### ISSUE THREAD

**Issue by UltraSirius** [AJS-development/Ogar-unlimited#1274]
This A Update For A Issue Ogar Unlimited!

@AJS-development/ogar-unlimited-devs

Hello Fellows AJS Teams! I Just Wanted A Request For A New Update! 

**AI UPDATES REQUEST:**
1) Make The Bots ignore Viruses (Because I see Some Bots Eats Viruses)
2) Make The Bots Shoot Virus (Like a real Human Being Shooting It)
3) Improve The Bots (like A Real Human, Teaming,splitting)
4) Make The Bots Do Like (PopSplit,TrickSplit,CannonSplit And Other techniques That Agar.io Players do)
5) Make The Player Cells/Bots Spawn At The Safe Please We'rent They Are Near At The Bigger Cells.
6) And ALSO  Improve The Bots Teaming. (They are now teaming at Bots or humans, by giving mass for Bots, They Will team at u! And If u eat them! They will BETRAY/KILL U!)

**CORE UPDATES REQUEST:**
1) Splitting Must be exact by a real agar.io
2) Cells Must be exact by a real agar.io
3) Skins Must be exact by a real agar.io
4) Add a Level and XP System just like In agar.io.io (Update it from play.ogarul.tk)
(BECAUSE I WANT TO INSULT MONEYCLIP A.K.A MiniClip)
----------------------------------------------------------------------------------------------------------                      
4) Don't Need to add Like Retina Or Somthing! add some old junks that came for Ogar Or Ogar Plus
5) Improve Some Stability From CORE JavaScript

**MODULES UPDATES REQUEST: **
1) Imrpove Some Stability From MODULES JavaScript

I Hope Some Staff Read This! because i always thinking those UPDATES! :))) PLEASE READ THIS, AND IF U READ THIS AJS Staff. Thank You For Reading :)

**Comment by AgarioIsBest1602** [comment]
Wow yes i like it! I hate bots act like they are worse than a noob

**Comment by AgarioIsBest1602** [comment]
But... You can change the settings of the bots in config.ini and in there u can make the bots a little bit smarter

**Comment by Agarian** [comment]
By `"smoother cells"` do you mean cell collisions? 
And `"smoother skins"`?

**Comment by UltraSirius** [comment]
yes cell collision. i just wanted the cells more like a real agar.io. and @AgarioIsBest1602 i am requesting for a new Update for the bots they need more improvement for the bots minds

**Comment by raphyjb** [comment]
could be amazing to have exact copy of agario. No lag, splitting is same (already done for splitting) ect..

**Comment by UltraSirius** [comment]
And Also update our play.ogarul.tk and add some level up thingy :) with only above level 100!

**Comment by LegitSoulja** [comment]
Nope. That's not gonna happen, unless they client gets re-written. What I can do is a skin unlocked, and a coin system. Which then you can earn coins from playing, which you can buy/unlock skins. 

Displaying the most top and richest player on site.

**Comment by UltraSirius** [comment]
Sure Why Not? lets use coin system!

**Comment by Agarian** [comment]
Please no coin system, we don't wanna be like miniclip!

**Comment by fayizan** [comment]
@LegitSoulja We need it..!!!

**Comment by Agarian** [comment]
:disappointed: :gun:

**Comment by LegitSoulja** [comment]
For AJS members, you can bypass the coin system if you want. There will be soon account registrations, and a coin system.

If you don't want to participate in the coin system then don't install the plugin.

**Comment by UltraSirius** [comment]
I will Participate But how do i bypass the coin system?

**Comment by LegitSoulja** [comment]
You won't make money, nor people won't by coins but they can earn coins depending on there game time and high score. Each server score will be calculated into one which then there will be a top player or maybe a player of the day reward.

I may just give away some rewards for winners, or maybe allow people to bet actual money to become first, and the winner takes it all.

There's a lot of things I actually want to do with this, but I'm going to need you guys help on which features should, and should not be added.

Same thing with ip bots for @raphyjb. Server owners are allowed to create there own site shop which I have been developing for a while, which they will have a custom link to there store, which users could then buy bots for a certain amount of hours. 

There's sooo many things I want to do with this project, but I'll just leave the thinking and future ideas up to everyone here.

**Comment by UltraSirius** [comment]
We Dont be like Miniclip but just dont get a Money thingy. lets say. if a player just kill a one Player he receives +25 Coins To Buy Powerups Or Unique Skins (Not Belong To MINICLIP) And add Some Like killstreak up too 50! every cell kill 5 players they will get a 5 Killstreak with a Bonus of 10+ Coins

**Comment by LegitSoulja** [comment]
We barely even use miniclip skins, since we kindof have our owns. I was saying, if there was money involved, you **won't** buy skins, but you can bet, or either win money. So technically, nobody isn't making money.

To what @raphyjb wants the the minions, he wants custom shops to allow people for his server buy bots for a certain amount of hours. Not all servers will have these things, which everything will be a plugin. It's your decision if you want to have people buy bots, or its up to people to buy bots for a server. Technically, with this. Anyone can get bots instead of waiting for minions with random minions (**GiveMeMinions**).

**Comment by UltraSirius** [comment]
Yes Your Right. But Like u said We not be Like Miniclip. but we can Just Only have a Changes For OgarUL. yes Ogarul Is Different From Miniclip (agar.io), But Players Are Issuing Bugs,Glitches And Lags, Thats Why I am Suggesting, We Can Add A Minimap For A Guide. A Tutorial. And add Some Guide For Hosting So If Anyone wanna host they Have some  Guide To Read. 

In Agar.io Structure. I am Saying we Must change the Structure Of the Cells,Skins And The Client. Reducing Lag Is Our Common Problem. Coin System Is not Needed because we're different From Miniclip Like i said (BECAUSE I WANT TO INSULT MONEYCLIP A.K.A MiniClip) we can tip to each other. (If u Want) but guess I've say Anything For this Suggestion.

**Comment by LegitSoulja** [comment]
Well, only if I knew a little bit more about canvas API, and only if the person who wrote the client made it more understandable than a mess. 

Me, myself even tried to add s cell border to canvas. I tried a few methods I knew, like line(moveTo) and tried to draw a transparent rectangle with a stroke, and neither seemed to even take effect. So I just gave up on it. 

Only if I knew more about it, I would make it better. But I'm too lazy to even try to understand, since working with canvas isn't a thing that I really use in daily life, and I don't think that I'll ever need it. I was thinking about re-writing it to get a better understand, but that's way too much.
